src/hooks/__tests__/use-fetch.test.ts

```diff
@@ -0,0 +1,103 @@
+import { renderHook, waitFor } from "@testing-library/react";
+import useFetch from "../use-fetch";
+
+describe("useFetch", () => {
+  test("should return is loading state while fetch is in progress", async () => {
+    const { result } = renderHook(() => useFetch(() => Promise.resolve(), ["deps"]));
+
+    await waitFor(() => {
+      expect(result.current[0]).toBe(undefined);
+      expect(result.current[1]).toBe(true);
+      expect(result.current[2]).toBe(undefined);
+    });
+  });
+
+  test("should return is loading state while fetch is completed", async () => {
+    const { result } = renderHook(() => useFetch(() => Promise.resolve(), ["deps"]));
+
+    await waitFor(() => {
+      expect(result.current[0]).toBe(undefined);
+      expect(result.current[1]).toBe(false);
```

Comment:
the above test is exactly like this one, but `result.current[1]` is expected to be true in there. what makes this one to be false?

Reply:
Only time actually. So the first runder it will be `true` and the second render it will be `false`